### PR TITLE
use fingerprints from database when mirror syncing

### PR DESCRIPTION
**Pull request by wolfv**
@btel let me know what you think

### CODE REVIEW

File: quetz/tasks/mirror.py [hunk 1]
```diff
@@ -184,40 +185,37 @@ def _func(package_name, metadata):
...
+
+        if package_fingerprints is None:
+            package_versions = (
+                dao.db.query(PackageVersion)
+                .filter(PackageVersion.channel_name == channel_name)
+                .filter(PackageVersion.platform == platform)
+                .all()
+            )
+
+            print(
```
Comment (btel) [inline quetz/tasks/mirror.py:205]:
```suggestion
            log.debug(
```

File: quetz/tasks/mirror.py [hunk 2]
```diff
@@ -272,6 +274,41 @@ def initial_sync_mirror(
             version_stack.enter_context(method) for method in version_methods
         ]
 
+        update_batch = []
+        update_size = 0
+
+        def handle_batch(update_batch):
+            logger.debug(update_batch)
```
Comment (btel) [inline quetz/tasks/mirror.py:281]: log message would be useful

File: quetz/tasks/mirror.py [hunk 3]
```diff
@@ -272,6 +274,41 @@ def initial_sync_mirror(
...
+            if not update_batch:
+                return False
+
+            remote_packages = []
+            try:
+                remote_packages.append(remote_repository.open(path))
```
Comment (btel) [inline quetz/tasks/mirror.py:287]: where does `path` come from? I thought you might iterate over paths in update_batch, but I can't see the iteration

File: quetz/tasks/mirror.py [hunk 4]
```diff
@@ -292,30 +329,17 @@ def initial_sync_mirror(
...
-                if not skip_errors:
-                    raise exc
+            if len(update_batch) >= max_batch_length or update_size >= max_batch_size:
+                logger.debug(f"Executing batch with {update_size}")
+                any_updated = handle_batch(update_batch)
+                update_batch = []
+                update_size = 0
+
+        # handle final batch
+        any_updated = handle_batch(update_batch)
```
Comment (btel) [inline quetz/tasks/mirror.py:342]: probably you want to do:
```suggestion
        any_updated = any_updated or handle_batch(update_batch)
```

File: quetz/tasks/mirror.py [hunk 5]
```diff
@@ -292,30 +329,17 @@ def initial_sync_mirror(
...
-                logger.error(
-                    f"could not process package {package_name} from channel"
-                    f"{channel_name} due to error {exc} of "
-                    f"type {exc.__class__.__name__}"
-                )
-                if not skip_errors:
-                    raise exc
+            if len(update_batch) >= max_batch_length or update_size >= max_batch_size:
+                logger.debug(f"Executing batch with {update_size}")
+                any_updated = handle_batch(update_batch)
```
Comment (btel) [inline quetz/tasks/mirror.py:337]:
```suggestion
                any_updated = any_updated or handle_batch(update_batch)
```

File: quetz/tasks/mirror.py (hunk 3 above)
Comment (wolfv) [reply quetz/tasks/mirror.py:287]: phew!

File: quetz/tasks/mirror.py [hunk 6]
```diff
@@ -292,30 +329,17 @@ def initial_sync_mirror(
             else:
                 logger.debug(f"updating package {package_name} form {arch}")
 
-            try:
-                remote_package = remote_repository.open(path)
-            except RemoteServerError:
-                logger.error(f"remote server error when getting a file {path}")
-                continue
+            update_batch.append(path)
+            update_size += metadata['size']
```
Comment (btel) [inline quetz/tasks/mirror.py:333]: you need to add the size to fixtures

File: quetz/tasks/mirror.py (hunk 4 above)
Comment (wolfv) [reply quetz/tasks/mirror.py]: I am using bitwise operator now (`|=`)

File: quetz/tasks/mirror.py (hunk 6 above)
Comment (wolfv) [reply quetz/tasks/mirror.py]: I've added a default value for now. That might be more robust anyways?

File: quetz/tasks/mirror.py (hunk 2 above)
Comment (wolfv) [reply quetz/tasks/mirror.py]: Done

**Pull request merged**